TTS/engine_wrapper.py

```diff
@@ -63,9 +65,13 @@ def run(self) -> Tuple[int, int]:
             self.call_tts("posttext", self.reddit_object["thread_post"])
 
         idx = None
-        for idx, comment in track(enumerate(self.reddit_object["comments"]), "Saving..."):
+        for idx, comment in track(
+            enumerate(self.reddit_object["comments"]), "Saving..."
+        ):
             # ! Stop creating mp3 files if the length is greater than max length.
             if self.length > self.max_length:
+                self.length -= self.last_clip_length
+                idx -= 1
```

Comment:
it won't work like that, break will end for loop

try this
```suggestion
        last_clip_subtracted = False
        for idx, comment in track(
            enumerate(self.reddit_object["comments"]), "Saving..."
        ):
            # ! Stop creating mp3 files if the length is greater than max length.
            while self.length > self.max_length:
                if not last_clip_subtracted:
                    self.length -= self.last_clip_length
                    idx -= 1
                    continue
                break
            else:
```
put remaining code of a function in `else` and put `break` on the same level as `else`